File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 1/8/2026, 5:27:46 p.m.
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxCur = 0
        maxSoFar = 0

        for i in range(1, len(prices)):
            maxCur += prices[i] - prices[i-1]
            maxCur = max(0, maxCur)
            maxSoFar = max(maxCur, maxSoFar)

        return maxSoFar

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.c
// Last updated: 1/8/2026, 5:27:50 p.m.
int maxProfit(int* prices, int pricesSize) {
    int i, min = INT_MAX, max = INT_MIN;

    for(i = 0; i < pricesSize; i++){
        if(min > prices[i]) min = prices[i];
        if(max < prices[i] - min) max = prices[i] - min;;
    }

    return max;
}